### PR TITLE
[MDS-5658] CRR Submission Notification Trigger

### PR DESCRIPTION
**Pull request by isuru-aot**
## Objective 

[MDS-5658](https://bcmines.atlassian.net/browse/MDS-5658)

- Added scripts to add mapping between `emli_contact` and `compliance_article`
- Added new email formats for MS and CORE
- Based on the mine details and report details recipients are decided to notify

Flow diagram on how to prepare the email list
EmailSelectionFlow drawio

Screenshots of emails
Screenshot 2024-03-21 165941
Screenshot 2024-03-21 165930

### CODE REVIEW

File: migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql
Comment (isuru-aot) [inline migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql:1]: Creating a `xref` table of  `compliance_article` and `emli_contact` but named it as `mine_report_notification` as the purpose of it is only for notification. And it also has other columns `is_major_mine`, `is regional_mine` which needs to decide the notification ricipient.

File: migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    is_major_mine                               boolean default false             NOT NULL,
+    is_regional_mine                            boolean default false             NOT NULL,
+
+    FOREIGN KEY (compliance_article_id) REFERENCES compliance_article(compliance_article_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (contact_guid) REFERENCES emli_contact(contact_guid) DEFERRABLE INITIALLY DEFERRED,
+    PRIMARY KEY (compliance_article_emli_contact_xref_guid)
+);
+ALTER TABLE mine_report_notification OWNER TO mds;
+
+CREATE TEMP TABLE IF NOT EXISTS temp_article_section_email_mapping (
```
Comment (asinn134) [inline migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql:16]: Nice 👍

File: migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+	SELECT
+        ca.compliance_article_id,
+        ec.contact_guid, 
+        tmp.is_major,
+        tmp.is_regional
+        FROM temp_article_section_email_mapping tmp
+        INNER JOIN emli_contact ec 
+        ON tmp.email = ec.email
+        
+        INNER JOIN compliance_article ca
```
Comment (asinn134) [inline migrations/sql/V2024.03.18.11.11__add_report_notification_contacts.sql:274]: 👍

**Review by asinn134 — APPROVED** [review]
Nice work :)

File: services/core-api/app/api/mines/reports/models/mine_report_notification.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        if self.contact_guid:
+            contact = EMLIContact.find_EMLI_contact_by_guid(self.contact_guid)
+            return contact.email
+        return None
+
+    def __repr__(self):
+        return '<mine_report_notification %r>' % self.compliance_article_emli_contact_xref_guid
+
+    @classmethod
+    def find_contact_by_compliance_article(cls, _section, _sub_section, _paragraph, _sub_paragraph):
```
Comment (matbusby-fw) [inline services/core-api/app/api/mines/reports/models/mine_report_notification.py:37]: Nice work here!

**Review by matbusby-fw — APPROVED** [review]
🐎

**Pull request merged**